src/gui/transferlistfilterswidget.cpp

```diff
@@ -241,9 +241,10 @@ void StatusFilterWidget::updateTorrentStatus(const BitTorrent::Torrent *torrent)
 {
     TorrentFilterBitset &torrentStatus = m_torrentsStatus[torrent];
 
-    const auto update = [&torrentStatus](const TorrentFilter::Type status, const bool needStatus, int &counter)
+    const auto update = [torrent, &torrentStatus](const TorrentFilter::Type status, int &counter)
     {
         const bool hasStatus = torrentStatus[status];
+        const bool needStatus = TorrentFilter(status).match(torrent);
```

Comment:
Why does it need to match tags, category and not sure if hash is needed for this?
Other than that can't see any problems.

Reply:
> Why does it need to match tags, category and not sure if hash is needed for this?

It doesn't and no tags, categories and hashes are set to be matched.

Reply:
The main idea is to use the same way of filtering on both sides to avoid inconsistencies.

Reply:
BTW, I'm thinking about decoupling TorrentFilter into several independent parts.